### PR TITLE
Support CVAT 2.X deployment using helm

### PR DESCRIPTION
**Pull request by se-wo**
### Motivation and context
The current charts provided in the helm-charts directory are broken. The latest docker images require that Open Policy Agent is available. OPA was added to the docker-compose files but not to the helm charts. This pull requests fixes this problem.
Moreover the templates were updated to support newer kubernetes versions. APIs like networking.k8s.io/v1beta1 and extensions/v1beta1 no longer exist in kubernetes versions 1.22 onwards. This pull request also adds support for networking.k8s.io/v1. Further, I fixed a small mistake where frontend templates used backend labels instead of their own.
Multiple issues like #4438 and #4192 are looking for a solution to CVAT 2.X support.

### How has this been tested?
The helm charts were tested locally with minikube v1.25.2 running k8s v1.23.3. Charts were deployed using Helm 3.8

### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have added a description of my changes into [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [x] I have updated the [documentation](
  https://github.com/opencv/cvat/blob/develop/README.md#documentation) accordingly
- [ ] ~~I have added tests to cover my changes~~
- [x] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [x] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2022 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

**Comment by nmanovic** [comment]
@se-wo , thanks for the contribution!

File: helm-chart/README.md [hunk 1]
```diff
@@ -1,58 +1,88 @@
...
+    helm dependency update
+    ```
+
 4. (Optional) Install ingress of your choice (for example: <https://github.com/kubernetes/ingress-nginx>)
 5. (Optional) Create certificates for https (for example: <https://github.com/jetstack/cert-manager/>)
 6. (Optional) Create values.override.yaml and override there parameters you want
 7. Change postgresql password as described below
-8. Add ingress to values.override.yaml(example also below)
-7. Deploy cvat using command below
+8. Create a rules.tar.gz archive containing at least all rego files in /cvat/apps/iam/rules not containing test in their name
```
Comment (azhavoro) [inline helm-chart/README.md:19]: Could you add a shell command for this? Also please add where this archive should be placed.

**Review by azhavoro — CHANGES_REQUESTED** [review]
@se-wo  thank you for the your contribution,
Generally I'm OK with changes, but could you please fix Markdown linter warning and add minor note about archive?

**Comment by se-wo** [comment]
> @se-wo thank you for the your contribution, Generally I'm OK with changes, but could you please fix Markdown linter warning and add minor note about archive? Also could you please add a note to the changelog, I think it would be helpful to the community.

Can you give me a hint on what the linter issue is or what linter style is used? I am only able to see that the linter Github action failed. I fixed everything the default VS Code linter reported to me before opening this pull request.

**Comment by se-wo** [comment]
@azhavoro added a changelog entry. Updated the readme with information about where to create the rules.tar.gz. Also added the shell command that I used to create the archive.

File: helm-chart/templates/cvat_opa/config.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  labels:
+    {{- include "cvat.labels" . | nindent 4 }}
+    app: cvat-app
+    tier: opa
+{{- if semverCompare ">=1.21-0" .Capabilities.KubeVersion.GitVersion }}
+immutable: true
+{{- end }}
+binaryData:
+  rules.tar.gz: |-
+    {{ .Files.Get "rules.tar.gz" | b64enc }}
```
Comment (zatkowich) [inline helm-chart/templates/cvat_opa/config.yml:16]: can you please add the file?

**Comment by Goseokbin** [comment]
@se-wo  Hi, i tried to deploy cvat with your branch. i checked to pods are running(front,backend,opa, redis,postgres)  and access frontend(expose with nodePort). but i have some error 'Could not check authorization on the server'(in console : TypeError: Cannot use 'in' operator to search for 'in' in..) . how to solve it?

**Comment by randombenj** [comment]
Thanks for the effort! I deployed my chart from this branch, and it works like a charm :tada:

**Comment by azhavoro** [comment]
@se-wo LGTM, could you please fix linter warning, I've attached the report
[remark_report (1).zip](https://github.com/openvinotoolkit/cvat/files/8348512/remark_report.1.zip)

**Comment by se-wo** [comment]
@azhavoro Linter error should be fixed. Was only a single line that exceeded 120 characters. Have split it into multiple lines.

**Review by azhavoro — APPROVED** [review]

**Review by nmanovic — APPROVED** [review]
@se-wo , Thanks for the contribution!

**Pull request merged**